Source/common/Keychain.h

```diff
@@ -0,0 +1,81 @@
+/// Copyright 2025 North Pole Security, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#ifndef SANTA__COMMON__KEYCHAIN_H
+#define SANTA__COMMON__KEYCHAIN_H
+
+#include <Foundation/Foundation.h>
+#include <Security/Security.h>
+
+#include <memory>
+
+#include "absl/status/status.h"
+#include "absl/status/statusor.h"
+
+namespace santa {
+
+namespace keychain_utils {
+
+// Validation helpers with semi-arbitrary length checks.
+bool IsValidServiceName(NSString *service);
+bool IsValidAccountName(NSString *account);
+bool IsValidDescription(NSString *description);
+
+absl::Status SecurityOSStatusToAbslStatus(OSStatus status);
+
+}  // namespace keychain_utils
+
+class KeychainItem;
```

Comment:
Why not declare KeychainItem first?

Reply:
Personal preference. When reading headers it makes more sense to me to see the "top level" objects you interact with first. Having KeychainItem first feels off to me since it's not something you construct manually.